src/util/timer.js

```diff
@@ -90,6 +90,25 @@ class Timer {
     timeElapsed () {
         return this.nowObj.now() - this.startTime;
     }
+
+    /**
+     * Call a handler function after a specified amount of time has elapsed.
+     * @param {function} handler - function to call after the timeout
+     * @param {number} timeout - number of milliseconds to delay before calling the handler
+     * @returns {number} - the ID of the new timeout
+     */
+    setTimeout (handler, timeout) {
+        return global.setTimeout(handler, timeout);
```

Comment:
`global` only exists in Node. Browser environments have `window` and `self`.

You could `typeof setTimeout !== 'undefined'` for it's existence. That'll work in any JS environment.

Reply:
```js
if (typeof setTimeout !== 'undefined') {
    return setTimeout(handler, timeout);
} else {
    // something else
}
```

Reply:
I should clarify `global` will work in browsers but only because webpack is polyfilling global as window for us.

Reply:
Webpack automatically converts `global` into `window` when targeting browsers -- or, rather, Webpack always provides a value for `global` and in the browser case it's generally `window`.

Anyway, `global` is only really here to make it explicit that I'm not making a recursive call. The function would work just as well without `global` or `self` or `window`, it would just look weird at first glance:
```js
setTimeout (handler, timeout) {
        return setTimeout(handler, timeout);
}
```

Reply:
> it would just look weird(er) at first glance:

I like this point.

> converts `global` into `window` when targeting browsers

webpack wraps your code in a second closure that depends on [webpack/buildins/global.js](https://github.com/webpack/webpack/blob/master/buildin/global.js). It notices the keyword and adds that wrapper. It also does that for `Buffer`, `process`, and I think some others.

```js
// the wrapping function stored in webpack's module array
function (module, exports, require) {
  (function (global) {
    // your module code here
  }(require('webpack/buildins/global'));
}
```

webpack is smart enough to not do this if `global` or any of the others are defined in a scope the variable name can reach in the module.